Update for 'netcdf'.

skosukhin opened the pull request:
This is an update for NetCDF.

There are several questions that need to be answered before this gets merged:

1. There was a workaround introduced by @adamjstewart in #1553. Do we still need it? I tried to run the commands from the discussion and they seem to work.
2. What do we do with options 'parallel4' and 'pnetcdf'? Should it be just one option 'mpi' that enables both parallel I/O features? If not, what should be the default values for these variants? Both set to True? See also #2377.
3. Should the package depend on 'mpi'? From one hand, we should run the MPI wrapper. From another hand, netcdf itself does not need mpi and `self.spec['mpi'].mpicc` is set for hdf5/pnetcdf anyway and we can use it.

davydden commented on var/spack/repos/builtin/packages/netcdf/package.py:
```diff
@@ -77,18 +80,49 @@ class Netcdf(AutotoolsPackage):
+
+    # High-level API of HDF5 1.8.9 or later is required for netCDF-4 support:
+    # http://www.unidata.ucar.edu/software/netcdf/docs/getting_and_building_netcdf.html
+    # TODO: change to 'hdf5+hl' when/if https://github.com/LLNL/spack/pull/5790
```
when/if is merged ?

skosukhin replied:
There is no 'hl' variant for hdf5 in the develop branch. The referenced PR introduces it.

davydden replied:
i know, i am saying you miss `is merged` in the sentence.

skosukhin replied:
Ok, done.

davydden commented on var/spack/repos/builtin/packages/netcdf/package.py:
```diff
@@ -104,91 +138,91 @@ def patch(self):
-            LDFLAGS.append("-L%s/lib"      % spec['parallel-netcdf'].prefix)
+            # undefined reference to `SSL_CTX_use_certificate_chain_file
+            curl = self.spec['curl']
+            curl_libs = self.spec['curl'].libs
```
`curl_libs = curl.libs`  (since you introduce `curl` above)

skosukhin replied:
Fixed.

skosukhin commented:
I can't compile with `+cdmremote` (even if I try it on the develop branch of Spack). It works only if `--disable-netcdf4` but it's not what we want. Could somebody try it? Should we keep this variant since it was removed starting version 4.4 anyway?

skosukhin commented:
I have managed to reproduce the problem described in #1553. Looks like `packages.yaml` gets accounted for in a wrong time. Added comments and replaced the workaround with `conflicts()` statements.

adamjstewart commented:
> There was a workaround introduced by @adamjstewart in #1553. Do we still need it? I tried to run the commands from the discussion and they seem to work.

There have been no changes to the concretizer as far as I know, so I'm going to say yes.

skosukhin commented:
I would say that this is ready for merging.
